src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -182,7 +196,7 @@ internal List<TResult> ToList<TResult>(Func<TKey, IEnumerable<TElement>, TResult
             return _count;
         }
 
-        public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
+        public IEnumerator<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
```

Comment:
We could, but it's a tiny gain for another type. While adding types to linq for small gains is pretty much my hobby these days, I still draw the line somewhere!

Another approach would be to make the iterator explicit, and have a constructor that put it straight into enumerator state. While that's more code, it wouldn't add to the produced IL as much.

Consider that cookie licked :wink: